src/index.ts

```diff
@@ -530,7 +530,24 @@ export function useState<S>(
             RootPath,
             () => setValue({ state: value.state }),
             value.state);
-        React.useEffect(() => () => value.state.destroy(), []);
+        const renders = React.useRef(0);
+
+        React.useMemo(function () {
+            return renders.current += 1;
+        }, [value.state]);
+
+        React.useEffect(() => {
+            const capture = renders.current;
+
+            return () => {
+                if (capture !== renders.current) {
+                    // do not destroy state if a third party (eg: react-fast-refresh) has restored it on re-render
```

Comment:
Please add link in comments to this page: https://github.com/apollographql/apollo-client/issues/5870#issuecomment-689098185
very useful